src/tesla_ver/bubble_chart/bubble_chart.py

```diff
@@ -62,8 +58,15 @@ def extract_mdata(df, x_column_name):
             }
 
         context = pa.default_serialization_context()
-        df = context.deserialize(redis_manager.redis.get("data_numeric"))
-        redis_manager.redis.flushdb()
+
+        if redis_manager.redis.exists("data_numeric"):
+            df = context.deserialize(redis_manager.redis.get("data_numeric"))
+            redis_manager.redis.flushdb()
+        else:
+            # Because of the need to return data matching all the different areas, displaying an error message
```

Comment:
what situation would this come up in?

Reply:
In case users navigate to the graphing screen right away, and try to start a graph without any data loaded, the code would raise a `PreventException` rather than error out with no message.  Ideally, this would show a prompt to go to the data loading screen as a text prompt, but dash callbacks require that the inputs and outputs be defined ahead of time (their initial state needs to be set on app launch, and therefore the framework doesn't let you have conditional outputs).  Either I'd need to feed in a fake set of data, which would trigger the computationally expensive graphing callbacks (even though they could be hidden) as well as show an error message while keeping the graph and controls hidden, or I'd need to nest this in another callback which either shows an error, and then keeps the other callbacks from firing/shows no error and displays the graph and fires the associated callbacks.

My rationale for leaving this off primarily came down to code complexity -- at this juncture I feel like adding an additional layer of callbacks that fire the others conditionally is a material jump in code complexity and reduces readability/understandability enough that a visible error message if the user goes to the graphing screen before the data loading screen doesn't make sense for a v0.1 release.